.. _repository-pullfailed-rei2:

失敗と対応の例 #2 ：  コンフリクトあり
====================================================================================================

.. _repository-pullfailed-rei2-editfile:

ファイルを編集
--------------------------------------------------
:file:`conf.py` ファイル内の「project=」を変更します。変更後セーブしますが、コミットはしません。

.. image:: img/2020-07-25_21h05_55.png
   :scale: 65%

↓

.. image:: img/2020-07-25_21h06_22.png
   :scale: 65%

|

.. _repository-pullfailed-rei2-pull:

プル　→　エラー発生
--------------------------------------------------
プル対象ファイルと未コミットのファイルがかち合うためエラーが発生します。

#. リモートリポジトリの状態を確認

   コミット状況の確認です。自分（ riki ）がプッシュ後に新しい内容が追加されています。

   .. image:: img/2020-07-25_21h22_22.png
      :scale: 65%

   |

   詳細です。 :file:`docs/source/conf.py` がプッシュされています。 

   .. image:: img/2020-07-25_21h22_41.png
      :scale: 65%

   |

#. :guilabel:`プル` をクリック

   .. image:: img/2020-07-25_21h23_05.png
      :scale: 65%

   |

#. プル元を確認　→　:guilabel:`プル` をクリック

   .. image:: img/2020-07-25_21h23_33.png
      :scale: 65%

   |

#. プル中にエラー発生　→　:guilabel:`閉じる(C)` をクリック

   .. image:: img/2020-07-25_21h24_40.png
      :scale: 65%

   |

   エラーメッセージと翻訳結果です。

   Your local changes to the following files would be overwritten by merge: 

   docs/source/conf.py

   Please commit your changes or stash them before you merge.

   ↓

   以下のファイルに対するローカルでの変更は、マージによって上書きされます。

   docs/source/conf.py

   マージする前に変更をコミットするか、隠しておいてください。

   |

#. :guilabel:`キャンセル` をクリック

   .. image:: img/2020-07-25_21h25_52.png
      :scale: 65%

   |

#. プル終了

   プル中にエラーが発生したため、リモートリポジトリの内容は取り込まれていません。

   .. image:: img/2020-07-25_21h25_59.png
      :scale: 65%

   |

.. _repository-pullfailed-rei2-commit:

コミット
--------------------------------------------------
未コミットのファイルをコミットします。

#. コミット前の状態　→　:guilabel:`コミットされていない変更があります` をクリック

   .. image:: img/2020-07-25_21h34_12.png
      :scale: 65%

   |

#. 対象ファイルをステージングエリアに登録　→　:guilabel:`コミット` をクリック

   .. image:: img/2020-07-25_21h34_52.png
      :scale: 65%

   |

#. コメントを入力　→　:guilabel:`コミット` をクリック

   .. image:: img/2020-07-25_21h35_43.png
      :scale: 65%

   |

#. コミット終了

   .. image:: img/2020-07-25_21h36_00.png
      :scale: 65%

   |

.. _repository-pullfailed-rei2-repull:

再プル　→　コンフリクト発生
--------------------------------------------------

.. admonition:: コンフリクト（衝突）

   同じファイルの同じ箇所の修正がかち合った（コンフリクト、衝突）状態のこと。

#. :guilabel:`プル` をクリック

   .. image:: img/2020-07-25_21h45_39.png
      :scale: 65%

   |

#. プル元を確認　→　:guilabel:`プル` をクリック

   .. image:: img/2020-07-25_21h46_16.png
      :scale: 65%

   |

#. プル中にコンフリクトが発生　→　:guilabel:`閉じる(C)` をクリック

   .. image:: img/2020-07-25_21h46_48.png
      :scale: 65%

   |

#. プル終了

   プル中にコンフリクトが発生しましたが、ローカルリポジトリにリモートリポジトリの変更内容が取り込まれました。 ``！`` マークが付いているのがコンフリクトが発生したファイルです。

   .. image:: img/2020-07-25_21h47_17.png
      :scale: 65%

   |

.. _repository-pullfailed-rei2-reedit:

コンフリクトが発生したファイルの修正
--------------------------------------------------

#. コンフリクトが発生した :file:`docs/source/conf.py` ファイルを開く

   コンフリクトマーカー（衝突マーカ－）の確認　＝　コンフリクトが発生した部分が対象ごとに色分けして編みかけされています。

   .. image:: img/2020-07-25_21h48_32.png
      :scale: 65%

   |

#. 対応方法の候補から適切なものを選択

   | 次の 3 つの対応方法の候補が表示されています。
   | ・ 現在の変更を取り込む　＝　ローカルリポジトリの内容を取り込みます
   | ・ 入力側の変更を取り込む　＝　リモートリポジトリの内容を取り込みます
   | ・ 両方の変更を取り込む　＝　両リポジトリの内容を取り込みます
   | 今回は「現在の変更を取り込む」を選択します。

   .. image:: img/2020-07-25_21h49_03.png
      :scale: 65%

   |

#. 修正内容の保存

   選択した候補の内容だけが表示されています。他に修正が必要な箇所があれば、このタイミングで修正します。修正後はファイルをセーブします。

   .. image:: img/2020-07-25_21h49_35.png
      :scale: 65%

   |
